django/db/migrations/autodetector.py

```diff
@@ -799,9 +799,11 @@ def _generate_added_field(self, app_label, model_name, field_name):
         # You can't just add NOT NULL fields with no default or fields
         # which don't allow empty strings as default.
         preserve_default = True
+        time_fields = (models.DateField, models.DateTimeField, models.TimeField)
         if (not field.null and not field.has_default() and
                 not isinstance(field, models.ManyToManyField) and
-                not (field.blank and field.empty_strings_allowed)):
+                not (field.blank and field.empty_strings_allowed) and
+                not (isinstance(field, time_fields) and (field.auto_now or field.auto_now_add))):
```

Comment:
I wonder if we need the `isinstance` checks? Presumably some third-party fields might implement those options too. Maybe something like `getattr(field, auto_now, False) or ...` would be fine?